src/debug/dart_debug_impl.ts

```diff
@@ -721,23 +792,32 @@ export class DartDebugSession extends DebugSession {
 				reason = "breakpoint";
 
 				const breakpoints = event.pauseBreakpoints.map((bp) => thread.breakpoints[bp.id]);
-				const hasUnconditionalBreakpoints = !!breakpoints.find((bp) => !bp.condition && !bp.logMessage);
-				const conditionalBreakpoints = breakpoints.filter((bp) => bp.condition);
-				const logPoints = breakpoints.filter((bp) => bp.logMessage);
-
-				// Evalute conditions to see if we should remain stopped or continue.
-				shouldRemainedStoppedOnBreakpoint =
-					hasUnconditionalBreakpoints
-					|| await this.anyBreakpointConditionReturnsTrue(conditionalBreakpoints, thread);
-
-				// Output any logpoint messages.
-				for (const logPoint of logPoints) {
-					// TODO: Escape triple quotes?
-					const logMessage = logPoint.logMessage
-						.replace(/(^|[^\\\$]){/g, "$1\${") // Prefix any {tokens} with $ if they don't have
-						.replace(/\\({)/g, "$1"); // Remove slashes
-					const printCommand = `print("""${logMessage}""")`;
-					await this.evaluateAndSendErrors(thread, printCommand);
+				// When attaching to an already-stopped process, this event can be handled before the
```

Comment:
@stuartmorgan I'm not quite sure what's going on here. Is this that we're hitting breakpoints in the VM before the debug adapter knows about them?

Reply:
Right, specifically if you attach to a process that is already stopped on a breakpoint (you can test by either using the Observatory web UI, or by disabling the cleanup-on-disconnect code, attaching, detaching while stopped on a breakpoint, and then re-attaching). In that case, this code runs before your recently-added code that builds the breakpoint map.